### PR TITLE
Add in PEP700 simple API v1.1 support

### LINKED ISSUE #1482: Ensure bandersnatch implements pep700 fields
https://peps.python.org/pep-0700/

- The api-version must specify version 1.1 or later.
- A new versions key is added at the top level.
- Two new “file information” keys, size and upload-time, are added to the files data.

Keys (at any level) with a leading underscore are reserved as private for index server use. No future standard will assign a meaning to any such key.

### PR DESCRIPTION
**Pull request by cooperlees**
- Move simple module to generate API 1.1 (JSON changes only)
  - Since there is only additions we don't really break 1.0 so just moving to 1.1
  - Seems pypi did this too
  - PEP700: https://peps.python.org/pep-0700/

Tests:
- Fix tests for v1.1 new fields based on our mocked metadata

Fixes #1482

### CODE REVIEW

File: src/bandersnatch/simple.py [hunk 1]
```diff
@@ -229,6 +232,8 @@ def generate_json_simple_page(
                         self.digest_name: r["digests"][self.digest_name],
                     },
                     "requires-python": r.get("requires_python", ""),
+                    "size": r.get("size", -1),
```
Comment (pfmoore) [inline src/bandersnatch/simple.py:235]: A size of -1 isn’t technically allowed by PEP 700. What circumstances would result in the actual size not being known? IIRC, this was discussed as part of the PEP 700 thread, and the conclusion was that an index must know the size as it has to be able to serve the file.

**Review by pfmoore — CHANGES_REQUESTED** [review]

File: src/bandersnatch/simple.py (hunk 1 above)
Comment (cooperlees) [reply src/bandersnatch/simple.py:235]: Rightio - Thanks. I feel that should be in the PEP for future implementers if we want to uphold / enforce this. Not a thread about the PEP.

Banndersnatch only pulls from PyPI.org / warehouse, so if it always responds the size, then we'll always have the size to use.

So do we recommend I just pull size and raise a `KeyError` if it does not exist?

Comment (pfmoore) [reply src/bandersnatch/simple.py]: Sorry, I don’t know enough about bandersnatch to know how it works. Why not fall back to just reading the files size from the file system?

Comment (cooperlees) [reply src/bandersnatch/simple.py]: bandersnatch relies on metadata to workout what to download and basically drive itself today. I don't think the complexity of adding a stat to check the filesystem (be it POSIX or s3 etc.) is worth it. Especially as to believe the reported size we should probably need to checksum it as well, relying on the checksum existing in the metadata. There is a config option to stop on any error a user can set which would see this `KeyError` when set. I think we'll just move forward with that model that bandersnatch has used for years. I think we're overcomplicating here if we do more.

**Pull request merged**